### ISSUE TITLE
avg_pool2d decomposition ignores divisor_override on the global-average fast path, returning sum/(H*W) instead of sum/divisor_override

### ISSUE THREAD

**Issue by botify-service** [tenstorrent/tt-torch#1247]
### Summary

The `avg_pool2d` decomposition in `tt_torch/dynamo/experimental/xla_decompositions.py` takes a
global-average-pooling fast path that computes `input.mean(...)`. `mean` always divides by the
number of elements, so the `divisor_override` argument — which PyTorch defines as *replacing*
that divisor — is silently ignored whenever the fast path fires.

### The decomposition

```python
def avg_pool2d(
    input, kernel_size, stride=None, padding=0,
    ceil_mode=False, count_include_pad=True,
    divisor_override: Optional[int] = None,      # <-- accepted
):
    ...
    input_size = list(input.shape[-len(stride):])
    if stride == kernel_size == input_size and padding == [0, 0, 0, 0]:
        return input.mean(dim=[-2, -1], keepdim=True)   # <-- divisor_override never read

    return NotImplemented
```

`divisor_override` is in the signature and is never referenced in the body. The guard checks
`stride`, `kernel_size`, `padding` and the input size, but nothing excludes a call that also
passes `divisor_override`, so such a call matches the fast path and returns the plain mean.

PyTorch's definition (`torch.nn.functional.avg_pool2d`):

> `divisor_override` – if specified, it will be used as divisor, otherwise size of the pooling
> region will be used.

So the expected result is `sum / divisor_override`; the decomposition returns
`sum / (H * W)`.

### Divergence

All-ones input, so `sum` equals the element count. Fast path fires in every row because
`kernel_size == stride == input_size` and `padding == 0`:

| input | kernel = stride | `divisor_override` | decomposition | PyTorch | off by |
|---|---|---|---|---|---|
| `[1,1,4,4]` | 4 | `2` | `1.0000` | `8.0000` | **8×** |
| `[1,1,4,4]` | 4 | `8` | `1.0000` | `2.0000` | **2×** |
| `[1,1,8,8]` | 8 | `16` | `1.0000` | `4.0000` | **4×** |
| `[1,1,2,2]` | 2 | `1` | `1.0000` | `4.0000` | **4×** |
| `[1,1,4,4]` | 4 | `None` | `1.0000` | `1.0000` | control — matches |

The error factor is exactly `(H*W) / divisor_override`, so it is unbounded in either direction
rather than a small numeric drift. The control row shares the same path and passes, which
isolates the cause to the ignored argument.

`ceil_mode` is also unread, but it is harmless here: when `kernel_size == stride == input_size`
a single window covers the input exactly, so ceil and floor agree. `count_include_pad` is moot
because the guard requires `padding == 0`. `divisor_override` is the only one that changes the
result.

### Minimum reproducible case

```python
import torch

x = torch.ones(1, 1, 4, 4)

expected = torch.nn.functional.avg_pool2d(
    x, kernel_size=4, stride=4, padding=0, divisor_override=2
)
print(expected)     # tensor([[[[8.]]]])  -> sum(16) / 2

# through the tt-torch decomposition this returns tensor([[[[1.]]]]) -> sum(16) / 16
```

Control that must keep passing in the same run: the identical call with `divisor_override=None`
returns `1.0` from both.

### Expected

The fast path should either honour the override or decline to match when one is present. The
smallest correct change is to exclude it from the guard:

```python
-    if stride == kernel_size == input_size and padding == [0, 0, 0, 0]:
+    if (stride == kernel_size == input_size and padding == [0, 0, 0, 0]
+            and divisor_override is None):
         return input.mean(dim=[-2, -1], keepdim=True)
```

which sends those calls down the `NotImplemented` path already used for every other unsupported
configuration, so the default implementation handles them. Alternatively, scale the result:
`input.sum(dim=[-2,-1], keepdim=True) / divisor_override`.

### Why the current tests do not catch it

`divisor_override` is rarely used and the pooling tests exercise the default, where it is `None`
and dropping it is a no-op. Nothing in the test suite passes a non-default `divisor_override`
together with a whole-input kernel, which is the only combination that triggers this.

### Notes on verification

Source analysis of the decomposition plus exact arithmetic; not run on hardware, and none is
needed — the argument is simply not referenced in the function body, and the divergence factor
is `(H*W)/divisor_override` by construction. The table assumes an all-ones input so that the
numerator is the element count; any input gives the same ratio.

### Impact

Low frequency but unbounded magnitude, and silent: the compiled graph returns a plausibly shaped
tensor scaled by an arbitrary constant. `divisor_override` shows up in models that emulate a
different pooling normalisation than the window size, and a fixed multiplicative error on a
pooled feature map is the kind of thing that degrades accuracy without ever raising.
